app/assets/javascripts/components/autocomplete.js.jsx

```diff
@@ -0,0 +1,46 @@
+var Autocomplete = React.createClass({
+  getInitialState: function() {
+    return {
+      data: JSON.parse(this.props.data),
```

Comment:
You shouldn't have to serialize to JSON in the view and deserialize in the React component. This is already taken care of by the view component library.